src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts

```diff
@@ -72,27 +84,62 @@ class encryptionSettings {
             this.enabled,
             this.mode,
             this.allowUnencryptedBackupForEncryptedDatabase,
-            this.key
+            this.changeKeyRequest
         ], false, jsonUtil.newLineNormalizingHashFunction);
         
-        _.bindAll(this, "useEncryptionType");
+        _.bindAll(this, "setEncryptionType");
     }
     
     private initObservables() {
+        this.validationGroup = ko.pureComputed(() => {
+            
+            if (this.enabled() && this.mode() === 'UseDatabaseKey') {
+                return this.validationGroupWithoutKey;
+            }
+            
+            return this.enabled() ? this.validationGroupWithKey : this.validationGroupWithoutKey;
+        });
+
+        setupEncryptionKey.setupKeyValidation(this.key);
+        
         this.backupType.subscribe(backupType => {
             const dbIsEncrypted = this.encryptedDatabase();
             if (dbIsEncrypted) {
                 if (this.backupType() === "Snapshot") {
                     this.mode("UseDatabaseKey");
                 }
             } else {
-                if (this.backupType() === "Backup") {
+                // db not encrypted
+                if (this.backupType() === "Backup" && this.enabled()) {
                     this.mode("UseProvidedKey");
+
+                    if (!this.originalKey()) {
+                        return this.encryptionSection.generateEncryptionKey();
+                    }
                 }
             }
         });
         
-        this.key.subscribe(() => this.keyConfirmation(false));
+        this.mode.subscribe(mode => {
+            if (this.encryptedDatabase() && mode === 'UseDatabaseKey') {
+                this.changeKeyRequest(false); // imho - better to clear this checkbox
+            }
+            
+            if (this.encryptedDatabase() && mode === 'UseProvidedKey' && !this.key()) {
+                return this.encryptionSection.generateEncryptionKey();
+            }
+        });
+        
+        this.key.subscribe(() => {
+            this.encryptionSection.syncQrCode();
+            this.keyConfirmation(false);
+        });
+
+        const dbName = ko.pureComputed(() => {
+            const db = activeDatabaseTracker.default.database();
```

Comment:
it is ugly side effect, can we pass this from external ?

Reply:
ok - done